android/src/main/kotlin/com/bbflight/background_downloader/BDPlugin.kt

```diff
@@ -378,16 +376,19 @@ class BDPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
             val startByte = if (args[3] is Long) args[3] as Long else (args[3] as Int).toLong()
             val eTag = args[4] as String?
             ResumeData(task, args[2] as String, startByte, eTag)
-        } else {
+        }
+        else {
             null
         }
+
+        bgChannelsByTask[task.taskId] = bgChannelsByContext[contextID]!!;
```

Comment:
You don't need this hashMap anymore, because if you have the context you can look up the backgroundChannel directly from the bgChannelsByContext